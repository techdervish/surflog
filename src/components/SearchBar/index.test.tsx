import {
  createShallowWrapper,
  expectWrappersTreeToMatchSnapshot,
  mocks,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('SearchBar component', () => {
  const mockFn = jest.fn();
  const wrapper = createWrapper({
    actions: {
      search: mockFn,
    },
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('renders Dropdown', () => {
    expect(wrapper.find('Dropdown')).toHaveLength(1);
  });

  it('calls onSearch on Dropdown change', () => {
    wrapper
      .find('Dropdown')
      .props()
      .onChange({ value: 'foo' });
    expect(mockFn).toBeCalled();
  });

  it('calls onSearch on Dropdown change', () => {
    wrapper.find('input').simulate('change', { target: { value: 'foo' } });
    expect(mockFn).toBeCalled();
  });
});
