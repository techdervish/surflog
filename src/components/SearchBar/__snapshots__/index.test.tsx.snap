// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders the expected snapshot 1`] = `
<div
  style={
    Object {
      "margin": "0 -16px",
    }
  }
>
  <div
    className="f4 pb3 black-70"
  >
    Search
     
    <Dropdown
      arrowClassName="source-dropdown-arrow"
      baseClassName="Dropdown"
      className="f4 dib v-btm bb b--gray fw3 black-50"
      controlClassName="none bn-ns pb0 fw4 mr2"
      menuClassName="source-dropdown-menu"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "all websites",
            "value": "all",
          },
          Object {
            "label": "Hackernews",
            "value": "hackernews",
          },
          Object {
            "label": "Twitter",
            "value": "twitter",
          },
          Object {
            "label": "Stackoverflow",
            "value": "stackoverflow",
          },
          Object {
            "label": "Reddit",
            "value": "reddit",
          },
          Object {
            "label": "Github",
            "value": "github",
          },
          Object {
            "label": "Youtube",
            "value": "youtube",
          },
        ]
      }
      placeholder="Select an option"
      value="all"
    />
     
  </div>
  <div>
    <input
      className="pa2 w-100 f5 fw3"
      onChange={[Function]}
      placeholder="enter keyword"
      style={
        Object {
          "outline": "none",
        }
      }
      type="text"
    />
  </div>
</div>
`;
